diff --git a/modules/charts_google/js/charts_google.js b/modules/charts_google/js/charts_google.js
index 5307bdb..3a02495 100755
--- a/modules/charts_google/js/charts_google.js
+++ b/modules/charts_google/js/charts_google.js
@@ -5,7 +5,7 @@
 
 /* global google */
 (function (Drupal, drupalSettings, once) {
-  Drupal.googleCharts = Drupal.googleCharts || { charts: [] };
+  Drupal.googleCharts = Drupal.googleCharts || { charts: {} };
 
   /**
    * Behavior to initialize Google Charts.
@@ -288,6 +288,7 @@
         options.colorAxis = { colors };
       }
       chart.draw(data, options);
+      Drupal.googleCharts.charts[chartId] = chart;
     };
   };
   