diff --git a/modules/charts_google/js/charts_google.js b/modules/charts_google/js/charts_google.js
index e99e27f..d93e24b 100755
--- a/modules/charts_google/js/charts_google.js
+++ b/modules/charts_google/js/charts_google.js
@@ -6,7 +6,7 @@
 
   'use strict';
 
-  Drupal.googleCharts = Drupal.googleCharts || {charts: []};
+  Drupal.googleCharts = Drupal.googleCharts || {charts: {}};
 
   /**
    * Behavior to initialize Google Charts.
@@ -155,6 +155,7 @@
         options['colorAxis'] = {colors: colors};
       }
       chart.draw(data, options);
+      Drupal.googleCharts.charts[chartId] = chart;
     };
   };
 
