diff --git a/src/Plugin/Block/GTranslateBlock.php b/src/Plugin/Block/GTranslateBlock.php
index 14f76db..92d1f7c 100644
--- a/src/Plugin/Block/GTranslateBlock.php
+++ b/src/Plugin/Block/GTranslateBlock.php
@@ -135,14 +135,13 @@ EOT;
                 }; break;
                 case 'dropdown': {
                     $block_content .= '<select onchange="doGTranslate(this);" id="gtranslate_selector" class="notranslate" aria-label="Website Language Selector">';
-                    $block_content .= '<option value="">Select Language</option>';
 
                     $i = 0;
                     foreach($languages as $lang => $lang_name) {
                         $flag_y = $flag_map_vertical[$lang];
 
                         if($settings->get('gtranslate_'.$lang)) {
-                            $block_content .= '<option value="'.$gtranslate_main_lang.'|'.$lang.'" style="'.($lang == $gtranslate_main_lang ? 'font-weight:bold;' : '').'background:url(\''.base_path().drupal_get_path('module', 'gtranslate').'/gtranslate-files/16l.png\') no-repeat scroll 0 -'.$flag_y.'px;padding-left:18px;">'.$lang_name.'</option>';
+                            $block_content .= '<option value="'.$gtranslate_main_lang.'|'.$lang.'" style="'.($lang == $gtranslate_main_lang ? 'font-weight:normal;' : '').'background:url(\''.base_path().drupal_get_path('module', 'gtranslate').'/gtranslate-files/16l.png\') no-repeat scroll 0 -'.$flag_y.'px;padding-left:18px;">'.$lang_name.'</option>';
                         }
 
                         $i++;
@@ -250,14 +249,13 @@ EOT;
                     }
 
                     $block_content .= '<select onchange="doGTranslate(this);" id="gtranslate_selector" class="notranslate" aria-label="Website Language Selector">';
-                    $block_content .= '<option value="">Select Language</option>';
 
                     $i = 0;
                     foreach($languages as $lang => $lang_name) {
                         if($settings->get('gtranslate_'.$lang)) {
                             $flag_y = $flag_map_vertical[$lang];
 
-                            $block_content .= '<option value="'.$gtranslate_main_lang.'|'.$lang.'" style="'.($lang == $gtranslate_main_lang ? 'font-weight:bold;' : '').'background:url(\''.base_path().drupal_get_path('module', 'gtranslate').'/gtranslate-files/16l.png\') no-repeat scroll 0 -'.$flag_y.'px;padding-left:18px;">'.$lang_name.'</option>';
+                            $block_content .= '<option value="'.$gtranslate_main_lang.'|'.$lang.'" style="'.($lang == $gtranslate_main_lang ? 'font-weight:normal;' : '').'background:url(\''.base_path().drupal_get_path('module', 'gtranslate').'/gtranslate-files/16l.png\') no-repeat scroll 0 -'.$flag_y.'px;padding-left:18px;">'.$lang_name.'</option>';
                         }
 
                         $i++;
