diff --git a/src/Core/Service/TarArchiveReader.php b/src/Core/Service/TarArchiveReader.php
index e4e522a..03823ae 100644
--- a/src/Core/Service/TarArchiveReader.php
+++ b/src/Core/Service/TarArchiveReader.php
@@ -137,7 +137,7 @@ class TarArchiveReader implements ArchiveReaderInterface {
 
       // Extract a directory.
       if ($header['typeflag'] == "5") {
-        if (!$this->createDir($header['filename'])) {
+        if (!$this->createDir($directory, dirname($filename))) {
           throw new BackupMigrateException(
             'Unable to create directory %filename',
             ['%filename' => $header['filename']]
@@ -146,7 +146,7 @@ class TarArchiveReader implements ArchiveReaderInterface {
       }
       // Extract a file/symlink.
       else {
-        if (!$this->createDir(dirname($header['filename']))) {
+        if (!$this->createDir($directory, dirname($filename)))) {
           throw new BackupMigrateException(
             'Unable to create directory for %filename',
             ['%filename' => $header['filename']]
@@ -231,23 +231,26 @@ class TarArchiveReader implements ArchiveReaderInterface {
    * @param mixed $directory
    *   The directory.
    *
+   * @param mixed $filename
+   *   The filename.
+   *
    * @return bool
    *   TRUE when successful, FALSE otherwise.
    */
-  private function createDir($directory) {
-    if ((@is_dir($directory)) || ($directory == '')) {
+  private function createDir($directory, $filename) {
+    if ((@is_dir($directory . $filename)) || ($filename == '')) {
       return TRUE;
     }
-    $parent = dirname($directory);
+    $parent = dirname($filename);
 
     if (
-      ($parent != $directory) &&
+      ($parent != $filename) &&
       ($parent != '') &&
-      (!$this->createDir($parent))
+      (!$this->createDir($directory, $parent))
     ) {
       return FALSE;
     }
-    if (@!mkdir($directory, 0777)) {
+    if (@!mkdir($directory . $filename, 0777)) {
       return FALSE;
     }
     return TRUE;
