diff --git a/src/Plugin/Field/FieldType/TimeRangeType.php b/src/Plugin/Field/FieldType/TimeRangeType.php
index 4b0ce99..cb0987b 100644
--- a/src/Plugin/Field/FieldType/TimeRangeType.php
+++ b/src/Plugin/Field/FieldType/TimeRangeType.php
@@ -68,8 +68,8 @@ class TimeRangeType extends FieldItemBase {
    * {@inheritdoc}
    */
   public function isEmpty() {
-    $from = $this->get('from')->getValue();
-    $to = $this->get('to')->getValue();
+    $from = $this->get('from')->getValue() ?? '';
+    $to = $this->get('to')->getValue() ?? '';
     return trim($from) === '' || trim($to) === '';
   }
 
diff --git a/src/Plugin/Field/FieldType/TimeType.php b/src/Plugin/Field/FieldType/TimeType.php
index e97cc98..31b2c35 100644
--- a/src/Plugin/Field/FieldType/TimeType.php
+++ b/src/Plugin/Field/FieldType/TimeType.php
@@ -83,7 +83,7 @@ class TimeType extends FieldItemBase {
    * {@inheritdoc}
    */
   public function isEmpty() {
-    $value = $this->get('value')->getValue();
+    $value = $this->get('value')->getValue() ?? '';
     return trim($value) === '';
   }
 
