diff --git a/core/modules/ckeditor/js/plugins/drupalmedia/plugin.es6.js b/core/modules/ckeditor/js/plugins/drupalmedia/plugin.es6.js
index 7a0c5a4213..913a2f9364 100644
--- a/core/modules/ckeditor/js/plugins/drupalmedia/plugin.es6.js
+++ b/core/modules/ckeditor/js/plugins/drupalmedia/plugin.es6.js
@@ -337,7 +337,7 @@
           const editButton = CKEDITOR.dom.element.createFromHtml(
             Drupal.theme('mediaEmbedEditButton'),
           );
-          embeddedMedia.getFirst().insertBeforeMe(editButton);
+          embeddedMedia.insertBeforeMe(editButton);
 
           // Make the edit button do things.
           const widget = this;
diff --git a/core/modules/ckeditor/js/plugins/drupalmedia/plugin.js b/core/modules/ckeditor/js/plugins/drupalmedia/plugin.js
index 8a8e1d574b..74e2508c56 100644
--- a/core/modules/ckeditor/js/plugins/drupalmedia/plugin.js
+++ b/core/modules/ckeditor/js/plugins/drupalmedia/plugin.js
@@ -197,7 +197,7 @@
           }
           embeddedMedia.setStyle('position', 'relative');
           var editButton = CKEDITOR.dom.element.createFromHtml(Drupal.theme('mediaEmbedEditButton'));
-          embeddedMedia.getFirst().insertBeforeMe(editButton);
+          embeddedMedia.insertBeforeMe(editButton);
           var widget = this;
           this.element.findOne('.media-library-item__edit').on('click', function (event) {
             var saveCallback = function saveCallback(values) {
